tools/rtpsrelay/RelayHandler.h

```diff
@@ -55,7 +55,13 @@ class VerticalHandler : public RelayHandler {
   VerticalHandler(ACE_Reactor* a_reactor,
                   const AssociationTable& a_association_table,
                   const ACE_Time_Value& lifespan,
-                  const OpenDDS::DCPS::RepoId& application_participant_guid);
+                  const OpenDDS::RTPS::RtpsDiscovery_rch& rtps_discovery,
+                  DDS::DomainId_t application_domain,
+                  const OpenDDS::DCPS::RepoId& application_participant_guid
+#ifdef OPENDDS_SECURITY
```

Comment:
This is getting a bit hard to manage.  I wonder if it would be better to have a consistent number of parameters and use the macro replacement just for its type (it could be `int` without security).  I think that would help cleaning up the calling code as well.